bigchaindb/consensus.py

```diff
@@ -162,6 +171,14 @@ def validate_transaction(bigchain, transaction):
                     raise exceptions.DoubleSpend(
                         'input `{}` was already spent'.format(fulfillment['input']))
 
+                txids.append(fulfillment['input']['txid'])
```

Comment:
The change you proposed will add a bit more code to create_tx but it will simplify validate_transaction which I think is a good tradeoff since we want validate_transaction to be as fast as possible